src/cdk-lib/bedrock/agent.ts

```diff
@@ -390,8 +394,7 @@ export class Agent extends Construct {
     super(scope, id);
     validatePromptOverrideConfiguration(props.promptOverrideConfiguration);
 
-    validateModel(props.foundationModel);
```

Comment:
Is it possible to pass an invalid Bedrock Agent's model? If so, what will happen?

Reply:
this just evaluated if the FM supports agents or not (as specified in models.ts), otherwise it threw an error at build time. Without this validation the error will be thrown at run-time. The idea is to refactor agents #731  and handle this appropriately according on whether it is an inference profile or a vanilla FM.